buffer/src/main/java/io/netty/buffer/ByteBufProcessor.java

```diff
@@ -16,10 +16,14 @@
 
 package io.netty.buffer;
 
-public interface ByteBufProcessor {
+/**
+ * Use {@link ByteVisitor} directly where interfaces permit.
+ */
+public interface ByteBufProcessor extends ByteVisitor {
```

Comment:
If I understood correctly, `ByteVisitor` is a renamed version of `ByteBufProcessor` and the rename is necessary because it now works against `ByteBuf`, `ByteString` (and `AsciiString`), right?

Then how about:

* Moving `ByteString`, `AsciiString` and `ByteVisitor` to netty-common (since it's not limited to writing a codec anymore)
* Deprecating `ByteBufProcessor` and let users use it wherever possible.
* Add the overridden methods of the methods who accepts `ByteBufProcessor` as a parameter, and deprecate the methods with `ByteBufProcessor`.